src/Data/List/Properties.agda

```diff
@@ -842,6 +842,19 @@ sum-++ (x ∷ xs) ys = begin
 ∈⇒∣product {n} {n ∷ ns} (here  refl) = divides (product ns) (*-comm n (product ns))
 ∈⇒∣product {n} {m ∷ ns} (there n∈ns) = ∣n⇒∣m*n m (∈⇒∣product n∈ns)
 
+nonZero-product : ∀ {ns} → All NonZero ns → NonZero (product ns)
+nonZero-product [] = _
+nonZero-product (nzn ∷ nzns) = m*n≢0 _ _ ⦃ nzn ⦄ ⦃ nonZero-product nzns ⦄
+
+∈⇒≤product : ∀ {n ns} → n ∈ ns → All NonZero ns → n ≤ product ns
+∈⇒≤product {n} {n ∷ ms} (here  refl) (_ ∷ nzms) =
+  ≤-trans (≤-reflexive (sym (*-identityʳ n)))
```

Comment:
I think using `≤-Reasoning` would make this proof (and the one below) much clearer.

Reply:
It might also permit a proof in which you don't force evaluation of the proof in the `here` pattern to be `refl`? I haven't checked this, though!

Reply:
I can do this, but I don't know if it's cleaner. The syntax for the cleanest version conflicts with the `≡-Reasoning` that's opened earlier in the module, so I have to redefine the equation part of the syntax to use it. Does that still sound worth it?

Reply:
Alternatively, use `m≤m*n` in the`here` branch, and `m≤n*m` in the `there` branch...?

As for the global opening of a reasoning module, and clashing with a local use of another, it's a fair point... we would now advise against it in the style guide, but this module carries way too much baggage (and for `Data.Nat.Properties` even more so). but I think the above suggestions finesse all this...?

Reply:
Okay, yeah, those do the job well.

Reply:
Maybe we should simply refactor `Data.Nat.Properties` to have lemmas of the form
```agda
m≤n⇒m≤o*n : {{NonZero o}} → m ≤ n → m ≤ o * n
m≤n⇒m≤n*o : {{NonZero o}} → m ≤ n → m ≤ n * o
```
and thereby localise all the `Nat`-based `≤-Reasoning` where it belongs... rather than in `Data.List.Properties`?

Reply:
Okay, I'll try that out.

Reply:
Okay, I've updated to take this approach and I think address the other comments.

I also noticed that there were already `<` versions of these m-n-o lemmas that are special cases, so I replaced their implementation with just the `≤` one.